src/electronic-ids/TLV.hpp

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2025 Estonian Information System Authority
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#pragma once
+
+#include "pcsc-cpp/pcsc-cpp.hpp"
+
+namespace electronic_id
+{
+
+struct TLV
```

Comment:
Needs a brief explanation that the constructor consumes the tag and length so that begin-end point to the value directly:
```suggestion
/**
 * Represents a single Tag-Length-Value structure used in DER-encoded eID card files.
 *
 * The constructor parses the tag and length from the provided byte range,
 * then adjusts its iterators so that `begin` and `end` reference only the value bytes.
 * If the TLV is empty, `operator bool()` returns false.
 */
struct TLV
```